src/compiler/moduleSpecifiers.ts

```diff
@@ -107,61 +109,63 @@ import {
 // Used by importFixes, getEditsForFileRename, and declaration emit to synthesize import module specifiers.
 
 const enum RelativePreference { Relative, NonRelative, Shortest, ExternalNonRelative }
-// See UserPreferences#importPathEnding
-const enum Ending { Minimal, Index, JsExtension }
 
 // Processed preferences
 interface Preferences {
     readonly relativePreference: RelativePreference;
-    readonly ending: Ending;
+    /**
+     * @param syntaxImpliedNodeFormat Used when the import syntax implies ESM or CJS irrespective of the mode of the file.
+     */
+    getAllowedEndingsInPrefererredOrder(syntaxImpliedNodeFormat?: SourceFile["impliedNodeFormat"]): ModuleSpecifierEnding[];

```

Comment:
*probably* should be `getAllowedEndingsInPreferredOrder`.
A prefererr sounds like a nightmare fiction of the http protocol.